Paco/src/com/pacoapp/paco/js/bridge/JavascriptEventLoader.java

```diff
@@ -107,13 +109,9 @@ public String getEventsForExperimentGroup() {
   public String getEventsByQuery(String criteriaQuery) throws JSONException, Exception {
     List<Event> events = null;
     String eventsJson = null;
-    SQLQuery sqlQueryObj = JsUtil.convertJSONToPOJO(criteriaQuery);
+    SQLQuery sqlQueryObj = QueryJsonParser.parseSqlQueryFromJson(criteriaQuery);
     if (sqlQueryObj != null) {
-      events = experimentProviderUtil.findEventsByCriteriaQuery(sqlQueryObj.getProjection(),
-                                                                sqlQueryObj.getCriteriaQuery(),
-                                                                sqlQueryObj.getCriteriaValue(),
-                                                                sqlQueryObj.getSortOrder(), sqlQueryObj.getLimit(),
-                                                                sqlQueryObj.getGroupBy(), sqlQueryObj.getHaving());
+      events = experimentProviderUtil.findEventsByCriteriaQuery(sqlQueryObj);
       eventsJson = FeedbackActivity.convertEventsToJsonString(events);
     } else {
       throw new RuntimeException("Empty JSON exception");
```

Comment:
We should return a json object, consistent with the server side EventQueryResultPair that has an error state and error message set.